### ISSUE TITLE
The Solution "Xbim.Geometry.Nuget" does not compile

### ISSUE THREAD

**Issue by martin-loucka** [xBimTeam/XbimGeometry#31]
Hi guys 

I am using Visual Studio 2012 and trying to build the sample solution "Xbim.Geometry.Nuget" 
Unfortunately, it is not possible to compile it . There are following Errors coming.

Do you know what can help?

Thank you
Martin

ERRORS:

Error	17	Metadata file 'C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Output\Debug\Xbim.ModelGeometry.Scene.dll' could not be found	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.Geometry.Engine.Tests\CSC	Xbim.Geometry.Engine.Tests

Error	18	Metadata file 'C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Output\Debug\Xbim.ModelGeometry.Scene.dll' could not be found	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.Geometry.Profiler\CSC	Xbim.Geometry.Profiler

Error	19	Metadata file 'C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Output\Debug\Xbim.ModelGeometry.Scene.dll' could not be found	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.Geometry.Regression\CSC	XbimRegression

Error	15	Metadata file 'C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Output\Debug\Xbim.Geometry.Engine.Interop.dll' could not be found	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.ModelGeometry.Scene\CSC	Xbim.ModelGeometry.Scene

Error	16	Metadata file 'C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Output\Debug\Xbim.Geometry.Engine.Interop.dll' could not be found	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.Geometry.Engine.Tests\CSC	Xbim.Geometry.Engine.Tests

Error	14	'Xbim::Geometry::XbimWire' : illegal member initialization: 'XbimWire' is not a base or member	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.Geometry.Engine\XbimWire.cpp	102	Xbim.Geometry.Engine.Interop

Error	13	'Xbim::Geometry::XbimVertex' : illegal member initialization: 'XbimVertex' is not a base or member	C:\_DEV\_IMRoomWallPOC\packages\XbimGeometry-master\XbimGeometry-master\Xbim.Geometry.Engine\XbimVertex.cpp	33	Xbim.Geometry.Engine.Interop

Error	12	'Xbim::Geometry::XbimSolidSet::_isSimplified': only static data members can be initialized inside a ref class or value type (XbimSolidSet.cpp)	c:\_dev\_imroomwallpoc\packages\xbimgeometry-master\xbimgeometry-master\xbim.geometry.engine\XbimSolidSet.h	40	Xbim.Geometry.Engine.Interop

Error	2	'Xbim::Geometry::XbimSolidSet::_isSimplified': only static data members can be initialized inside a ref class or value type (XbimSolid.cpp)	c:\_dev\_imroomwallpoc\packages\xbimgeometry-master\xbimgeometry-master\xbim.geometry.engine\XbimSolidSet.h	40	Xbim.Geometry.Engine.Interop

Error	3	'Xbim::Geometry::XbimSolidSet::_isSimplified': only static data members can be initialized inside a ref class or value type (XbimShellSet.cpp)	c:\_dev\_imroomwallpoc\packages\xbimgeometry-master\xbimgeometry-master\xbim.geometry.engine\XbimSolidSet.h	40	Xbim.Geometry.Engine.Interop

Error	6	'Xbim::Geometry::XbimSolidSet::_isSimplified': only static data members can be initialized inside a ref class or value type (XbimShell.cpp)	c:\_dev\_imroomwallpoc\packages\xbimgeometry-master\xbimgeometry-master\xbim.geometry.engine\XbimSolidSet.h	40	Xbim.Geometry.Engine.Interop

etc....

**Comment by CBenghi** [comment]
Hello Martin,
unfortunately we have upgraded the vc++ project to VS2013 and the c# to VS2015, so at the moment Geometry needs visuals studio 2015 with the C++ toolkit of VS2013.
Can I ask for what reason you are trying to compile? Most users could use the library through the nuget package available on nuget.org (https://www.nuget.org/packages/Xbim.Geometry/).
Best, 
Claudio

**Comment by martin-loucka** [comment]
Hello Claudio

The problem is that the Program crashes when the object XbimGeometryEngine is referenced... Could you please provide some example how to use it? 

Thanks 
Martin

**Comment by CBenghi** [comment]
Unfortunately I don't have a copy of VS2012 installed on any PC.
Can you try to test if the code at 
https://xbimteam.github.io/examples/creating-wexbim-file.html
works for you on VS2012?

**Comment by martin-loucka** [comment]
I have solved it. The library was running in a third program and the problem was there. Thank you anyway!

**Comment by lucasaue** [comment]
Hi,

I'm reopening this topic because I'm trying to build this solution on vc110 (VS 2012), but does not work.
I downgraded the code to a older version from c++, and the compilation works fine, on VS2012, but when I try to run the .exe, the xbim.geometry.engine still want the VS120 (VS 2013). 

I tryed to build from VS 2012, changing the Platform Toolset to VC110, and I tryed to force the build on vc110 on prompt to, by this code:

C:\Windows\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe D:\prg.2018\xBIM\XbimGeometry-master\Xbim.Geometry.Engine\Xbim.Geometry.Engine.vcxproj /p:VisualStudioVersion=11.0 /p:Configuration=Release

But, when I run the software, still ask me by the VC120.

There are something there I'm missing? Some thirds lib parts there are build on vc120?

I'm trying to do that because we have some others softwares that is older and uses vc110, so to do a clean setup, we do not want to use 2 vcredists
